hca/added_command.py

```diff
@@ -207,27 +212,103 @@ def _build_body_payload(cls, namespace):
             if payload_format == "body":
                 hierarchy = endpoint_info["options"][arg_name]["hierarchy"]
                 cls._add_arg(arg, body_payload, hierarchy)
         return body_payload
 
     @classmethod
-    def _get_auth_header(cls, real_header=True):
-        try:
-            credentials, project_id = auth.default(scopes=["https://www.googleapis.com/auth/userinfo.email"])
-
-            r = Request()
-            credentials.refresh(r)
-            r.session.close()
+    def _get_auth_header(cls, args, retry=False):
+        """
+        Called in run when a user calls an authenticated method. retry=True if authenticated method returns 401.
 
-            token = credentials.token if real_header else str(uuid.uuid4())
+        :param args: Dict of arguments for any given function.
+        :param retry: Boolean indicating if this is the second time trying to authenticate. If so, refresh token.
+        """
+        token = cls._get_access_token(args, retry)
+        return {'Authorization': "Bearer {}".format(token)}
 
-            return "Bearer {}".format(token)
+    @classmethod
+    def _get_access_token(cls, args, retry):
+        config = Config(Constants.TWEAK_PROJECT_NAME)
+        access_token = None
+
+        # kwargs access_token input
+        if 'access_token' in args:
+            if retry:
+                logging.info("Access token taken from kwargs invalid.")
+                raise ValueError("The access token you've supplied in the kwargs is not valid."
+                                 " Please refresh and try again. You may also run `hca login`"
+                                 " to get a more permanent hca configuration.")
+            else:
+                logging.info("Found access token in kwargs.")
+                access_token = args['access_token']
+
+        # Checking config
+        elif config.get('access_token', None):
+            logging.info(config['access_token'])
+            # There is a refresh token
+            if retry and config.get('refresh_token', None):
+                logging.info("The access token stored in {} is not valid."
+                             " Attempting with refresh token.".format(config.config_files[-1]))
+                credentials = oauth2client.client.OAuth2Credentials(
+                    access_token=None,
+                    client_id=config.client_id,
+                    client_secret=config.client_secret,
+                    scopes=["https://www.googleapis.com/auth/userinfo.email"],
+                    refresh_token=config.refresh_token,
+                    token_expiry=None,
+                    token_uri=config.token_uri,
+                    user_agent=None,
+                    revoke_uri=None
+                )
+                # Silence ResourceWarning from httplib2 socket being open for connection pooling.
+                warnings.simplefilter("ignore")
```

Comment:
This is not an appropriate command for a library to run. Let's figure out a way to go without it.